taichi/backends/dx/dx_device.cpp

```diff
@@ -0,0 +1,464 @@
+#include <unordered_map>
+
+#include "taichi/backends/dx/dx_device.h"
+#include "taichi/backends/dx/dx_api.h"
+#include "spirv_hlsl.hpp"
+
+namespace taichi {
+namespace lang {
+namespace directx11 {
+
+extern IDXGISwapChain *g_swapchain;
+
+static uint32_t g_alloc_id = 0;
+
+// Mapping from binding ID to Buffer
+static std::unordered_map<uint32_t, ID3D11Buffer *> g_binding2buf;
```

Comment:
These should be put inside DX11Device (similar to the Vulkan backend), we should refrain from using global variables

Reply:
Moved the binding to alloc ID map to DxResourceBinder.
Moved the alloc_id to Buffer and to UAV maps to DxDevice.